go data race during the test run on Mac

```
➜  image-scanner-operator git:(main) ✗ make test
test -s /Users/padlar/sandbox/image-scanner-operator/bin/controller-gen && /Users/padlar/sandbox/image-scanner-operator/bin/controller-gen --version | grep -q v0.11.1 || \
	GOBIN=/Users/padlar/sandbox/image-scanner-operator/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.11.1
/Users/padlar/sandbox/image-scanner-operator/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/Users/padlar/sandbox/image-scanner-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
test -s /Users/padlar/sandbox/image-scanner-operator/bin/setup-envtest || GOBIN=/Users/padlar/sandbox/image-scanner-operator/bin go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
KUBEBUILDER_ASSETS="/Users/padlar/sandbox/image-scanner-operator/bin/k8s/1.26.0-darwin-amd64" go test -race ./... -coverprofile cover.out
?   	github.com/statnett/image-scanner-operator/api/stas/v1alpha1	[no test files]
?   	github.com/statnett/image-scanner-operator/cmd	[no test files]
?   	github.com/statnett/image-scanner-operator/internal/config	[no test files]
ok  	github.com/statnett/image-scanner-operator/internal/controller	0.850s	coverage: 100.0% of statements
ok  	github.com/statnett/image-scanner-operator/internal/controller/stas	13.359s	coverage: 81.2% of statements
ok  	github.com/statnett/image-scanner-operator/internal/errors	0.444s	coverage: 42.9% of statements
?   	github.com/statnett/image-scanner-operator/internal/hash	[no test files]
ok  	github.com/statnett/image-scanner-operator/internal/metrics	1.726s	coverage: 88.1% of statements
?   	github.com/statnett/image-scanner-operator/internal/operator	[no test files]
ok  	github.com/statnett/image-scanner-operator/internal/pod	2.355s	coverage: 8.0% of statements
Running Suite: Resources Suite - /Users/padlar/sandbox/image-scanner-operator/internal/resources
================================================================================================
Random Seed: 1674740153

Will run 3 of 3 specs
•••==================
WARNING: DATA RACE
Read at 0x000003fb2c88 by goroutine 27:
  github.com/statnett/image-scanner-operator/internal/resources.glob..func3()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:54 +0x30
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111

Previous write at 0x000003fb2c88 by goroutine 120:
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2.1()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:49 +0x6c

Goroutine 27 (running) created at:
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:841 +0x12de
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSuiteNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:591 +0xe4a
  github.com/onsi/ginkgo/v2/internal.(*Suite).runAfterSuiteCleanup()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:523 +0x404
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:464 +0x6cb
  github.com/onsi/ginkgo/v2/internal.(*Suite).Run()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:110 +0x5c9
  github.com/onsi/ginkgo/v2.RunSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/core_dsl.go:296 +0x102a
  github.com/statnett/image-scanner-operator/internal/resources.TestAPIs()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:28 +0x55
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 120 (finished) created at:
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:47 +0x4c8
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111
==================
==================
WARNING: DATA RACE
Write at 0x00c00043eb50 by goroutine 27:
  sync/atomic.CompareAndSwapInt32()
      /usr/local/go/src/runtime/race_amd64.s:308 +0xb
  sync/atomic.CompareAndSwapInt32()
      <autogenerated>:1 +0x1e
  context.(*cancelCtx).cancel()
      /usr/local/go/src/context/context.go:401 +0x84
  context.WithCancel.func1()
      /usr/local/go/src/context/context.go:238 +0x4f
  github.com/statnett/image-scanner-operator/internal/resources.glob..func3()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:54 +0x3c
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111

Previous write at 0x00c00043eb50 by goroutine 120:
  context.WithCancel()
      /usr/local/go/src/context/context.go:236 +0x95
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2.1()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:49 +0x5b

Goroutine 27 (running) created at:
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:841 +0x12de
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSuiteNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:591 +0xe4a
  github.com/onsi/ginkgo/v2/internal.(*Suite).runAfterSuiteCleanup()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:523 +0x404
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:464 +0x6cb
  github.com/onsi/ginkgo/v2/internal.(*Suite).Run()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:110 +0x5c9
  github.com/onsi/ginkgo/v2.RunSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/core_dsl.go:296 +0x102a
  github.com/statnett/image-scanner-operator/internal/resources.TestAPIs()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:28 +0x55
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 120 (finished) created at:
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:47 +0x4c8
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111
==================
==================
WARNING: DATA RACE
Read at 0x00c00043eb70 by goroutine 27:
  context.(*cancelCtx).cancel()
      /usr/local/go/src/context/context.go:402 +0x9a
  context.WithCancel.func1()
      /usr/local/go/src/context/context.go:238 +0x4f
  github.com/statnett/image-scanner-operator/internal/resources.glob..func3()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:54 +0x3c
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111

Previous write at 0x00c00043eb70 by goroutine 120:
  context.WithCancel()
      /usr/local/go/src/context/context.go:236 +0x95
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2.1()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:49 +0x5b

Goroutine 27 (running) created at:
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:841 +0x12de
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSuiteNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:591 +0xe4a
  github.com/onsi/ginkgo/v2/internal.(*Suite).runAfterSuiteCleanup()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:523 +0x404
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:464 +0x6cb
  github.com/onsi/ginkgo/v2/internal.(*Suite).Run()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:110 +0x5c9
  github.com/onsi/ginkgo/v2.RunSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/core_dsl.go:296 +0x102a
  github.com/statnett/image-scanner-operator/internal/resources.TestAPIs()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:28 +0x55
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 120 (finished) created at:
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:47 +0x4c8
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111
==================
==================
WARNING: DATA RACE
Read at 0x00c00043eb58 by goroutine 27:
  sync/atomic.LoadInt64()
      /usr/local/go/src/runtime/race_amd64.s:206 +0xb
  sync/atomic.LoadPointer()
      <autogenerated>:1 +0x16
  context.(*cancelCtx).cancel()
      /usr/local/go/src/context/context.go:407 +0x104
  context.WithCancel.func1()
      /usr/local/go/src/context/context.go:238 +0x4f
  github.com/statnett/image-scanner-operator/internal/resources.glob..func3()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:54 +0x3c
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111

Previous write at 0x00c00043eb58 by goroutine 120:
  context.WithCancel()
      /usr/local/go/src/context/context.go:236 +0x95
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2.1()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:49 +0x5b

Goroutine 27 (running) created at:
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:841 +0x12de
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSuiteNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:591 +0xe4a
  github.com/onsi/ginkgo/v2/internal.(*Suite).runAfterSuiteCleanup()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:523 +0x404
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:464 +0x6cb
  github.com/onsi/ginkgo/v2/internal.(*Suite).Run()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:110 +0x5c9
  github.com/onsi/ginkgo/v2.RunSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/core_dsl.go:296 +0x102a
  github.com/statnett/image-scanner-operator/internal/resources.TestAPIs()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:28 +0x55
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 120 (finished) created at:
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:47 +0x4c8
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111
==================
==================
WARNING: DATA RACE
Write at 0x00c00043eb60 by goroutine 27:
  sync/atomic.StoreInt64()
      /usr/local/go/src/runtime/race_amd64.s:235 +0xb
  sync/atomic.StorePointer()
      /usr/local/go/src/runtime/atomic_pointer.go:51 +0x44
  context.(*cancelCtx).cancel()
      /usr/local/go/src/context/context.go:409 +0x14a
  context.WithCancel.func1()
      /usr/local/go/src/context/context.go:238 +0x4f
  github.com/statnett/image-scanner-operator/internal/resources.glob..func3()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:54 +0x3c
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111

Previous write at 0x00c00043eb60 by goroutine 120:
  context.WithCancel()
      /usr/local/go/src/context/context.go:236 +0x95
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2.1()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:49 +0x5b

Goroutine 27 (running) created at:
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:841 +0x12de
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSuiteNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:591 +0xe4a
  github.com/onsi/ginkgo/v2/internal.(*Suite).runAfterSuiteCleanup()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:523 +0x404
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:464 +0x6cb
  github.com/onsi/ginkgo/v2/internal.(*Suite).Run()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:110 +0x5c9
  github.com/onsi/ginkgo/v2.RunSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/core_dsl.go:296 +0x102a
  github.com/statnett/image-scanner-operator/internal/resources.TestAPIs()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:28 +0x55
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 120 (finished) created at:
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:47 +0x4c8
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111
==================
==================
WARNING: DATA RACE
Read at 0x00c00043eb68 by goroutine 27:
  context.(*cancelCtx).cancel()
      /usr/local/go/src/context/context.go:413 +0x164
  context.WithCancel.func1()
      /usr/local/go/src/context/context.go:238 +0x4f
  github.com/statnett/image-scanner-operator/internal/resources.glob..func3()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:54 +0x3c
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111

Previous write at 0x00c00043eb68 by goroutine 120:
  context.WithCancel()
      /usr/local/go/src/context/context.go:236 +0x95
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2.1()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:49 +0x5b

Goroutine 27 (running) created at:
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:841 +0x12de
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSuiteNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:591 +0xe4a
  github.com/onsi/ginkgo/v2/internal.(*Suite).runAfterSuiteCleanup()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:523 +0x404
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:464 +0x6cb
  github.com/onsi/ginkgo/v2/internal.(*Suite).Run()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:110 +0x5c9
  github.com/onsi/ginkgo/v2.RunSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/core_dsl.go:296 +0x102a
  github.com/statnett/image-scanner-operator/internal/resources.TestAPIs()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:28 +0x55
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 120 (finished) created at:
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:47 +0x4c8
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111
==================
==================
WARNING: DATA RACE
Read at 0x00c00043eb40 by goroutine 27:
  context.(*cancelCtx).cancel()
      /usr/local/go/src/context/context.go:421 +0x267
  context.WithCancel.func1()
      /usr/local/go/src/context/context.go:238 +0x4f
  github.com/statnett/image-scanner-operator/internal/resources.glob..func3()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:54 +0x3c
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111

Previous write at 0x00c00043eb40 by goroutine 120:
  context.WithCancel()
      /usr/local/go/src/context/context.go:236 +0x95
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2.1()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:49 +0x5b

Goroutine 27 (running) created at:
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:841 +0x12de
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSuiteNode()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:591 +0xe4a
  github.com/onsi/ginkgo/v2/internal.(*Suite).runAfterSuiteCleanup()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:523 +0x404
  github.com/onsi/ginkgo/v2/internal.(*Suite).runSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:464 +0x6cb
  github.com/onsi/ginkgo/v2/internal.(*Suite).Run()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:110 +0x5c9
  github.com/onsi/ginkgo/v2.RunSpecs()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/core_dsl.go:296 +0x102a
  github.com/statnett/image-scanner-operator/internal/resources.TestAPIs()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:28 +0x55
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 120 (finished) created at:
  github.com/statnett/image-scanner-operator/internal/resources.glob..func2()
      /Users/padlar/sandbox/image-scanner-operator/internal/resources/suite_test.go:47 +0x4c8
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /Users/padlar/code/pkg/mod/github.com/onsi/ginkgo/v2@v2.7.0/internal/suite.go:854 +0x111
==================


Ran 3 of 3 Specs in 5.372 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 0 Skipped
--- FAIL: TestAPIs (5.38s)
    testing.go:1312: race detected during execution of test
FAIL
coverage: 81.8% of statements
FAIL	github.com/statnett/image-scanner-operator/internal/resources	6.515s
ok  	github.com/statnett/image-scanner-operator/internal/trivy	1.063s	coverage: 37.3% of statements
ok  	github.com/statnett/image-scanner-operator/internal/yaml	0.754s	coverage: 100.0% of statements
FAIL
make: *** [test] Error 1
```

--- a/internal/resources/suite_test.go
+++ b/internal/resources/suite_test.go
@@ -1,7 +1,6 @@
 package resources
 
 import (
-	"context"
 	"testing"
 
 	. "github.com/onsi/ginkgo/v2"
@@ -9,7 +8,6 @@ import (
 	appsv1 "k8s.io/api/apps/v1"
 	batchv1 "k8s.io/api/batch/v1"
 	"k8s.io/client-go/kubernetes/scheme"
-	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
 	"sigs.k8s.io/controller-runtime/pkg/envtest/komega"
@@ -20,7 +18,6 @@ import (
 var (
 	k8sClient client.Client
 	testEnv   *envtest.Environment
-	cancel    context.CancelFunc
 )
 
 func TestAPIs(t *testing.T) {
@@ -29,8 +26,6 @@ func TestAPIs(t *testing.T) {
 }
 
 var _ = BeforeSuite(func() {
-	_, cancel = context.WithCancel(context.TODO())
-
 	By("bootstrapping test environment")
 	testEnv = &envtest.Environment{}
 	restCfg, err := testEnv.Start()
@@ -43,15 +38,9 @@ var _ = BeforeSuite(func() {
 	k8sClient, err = client.New(restCfg, client.Options{Scheme: scheme.Scheme})
 	Expect(err).NotTo(HaveOccurred())
 	komega.SetClient(k8sClient)
-
-	go func() {
-		defer GinkgoRecover()
-		_, cancel = context.WithCancel(ctrl.SetupSignalHandler())
-	}()
 })
 
 var _ = AfterSuite(func() {
-	cancel()
 	By("tearing down the test environment")
 	Expect(testEnv.Stop()).To(Succeed())
 })

